services/dynamic-sidecar/tests/unit/test_api_containers.py

```diff
@@ -410,57 +447,69 @@ def _expected_error_string(status_code: int) -> dict[str, Any]:
 
 async def test_outputs_watcher_disabling(
     test_client: TestClient,
+    mocked_port_key_events_queue_coro_get: Mock,
     mock_event_filter_enqueue: AsyncMock,
 ):
     assert isinstance(test_client.application, FastAPI)
     outputs_context: OutputsContext = test_client.application.state.outputs_context
     outputs_manager: OutputsManager = test_client.application.state.outputs_manager
     outputs_manager.task_monitor_interval_s = WAIT_FOR_OUTPUTS_WATCHER / 10
-    WAIT_FOR_EVENTS = outputs_manager.task_monitor_interval_s * 10
+    WAIT_PORT_KEY_PROPAGATION = outputs_manager.task_monitor_interval_s * 10
 
-    async def _create_file_in_random_dir_in_inputs() -> int:
+    async def _create_port_key_events() -> None:
         random_subdir = f"{uuid4()}"
 
         await outputs_context.set_file_type_port_keys([random_subdir])
-        await asyncio.sleep(WAIT_FOR_EVENTS)
+        await asyncio.sleep(WAIT_PORT_KEY_PROPAGATION)
 
         dir_name = outputs_context.outputs_path / random_subdir
-        dir_name.mkdir()
-
+        await mkdir(dir_name)
         async with aiofiles.open(dir_name / f"file_{uuid4()}", "w") as f:
             await f.write("ok")
 
-        dir_count = len(
-            [
-                1
-                for x in outputs_context.outputs_path.glob("*")
-                if not f"{x}".endswith(HIDDEN_FILE_NAME)
-            ]
-        )
-        await asyncio.sleep(WAIT_FOR_EVENTS)
-        return dir_count
-
-    CALLS_RECEIVED_BY_EVENT_FILTER = 3
+        async for attempt in AsyncRetrying(**_TENACITY_RETRY_PARAMS):
+            with attempt:
+                # check event was triggered
+                dir_event_set = {
+                    c.args[0]
+                    for c in mocked_port_key_events_queue_coro_get.call_args_list
+                    if c.args[0] == random_subdir
+                }
+                assert len(dir_event_set) == 1
+
+    def _assert_events_generated(*, expected_events: int) -> None:
+        events_set = {x.args[0] for x in mock_event_filter_enqueue.call_args_list}
+        assert len(events_set) == expected_events
+
+    # NOTE: for some reason the first event in the queue
+    #  does not get delivered the AioQueue future handling coro_get hangs
```

Comment:
hmm can you maybe explain what this means? you need to repeat something here?

Reply:
Let's have a look at it together, I can show you more easily.